homeassistant/components/nissan_leaf.py

```diff
@@ -0,0 +1,338 @@
+"""
+Support for the Nissan Leaf Carwings/Nissan Connect API.
+
+Please note this is the pre-2018 API, which is still functional in the US.
+The old API should continue to work for the forseeable future.
+
+Documentation has not been created yet, here is an example configuration block:
+
+nissan_leaf:
+  username: "username"
+  password: "password"
+  nissan_connect: false
+  region: 'NE'
+  update_interval: 30
+  update_interval_charging: 15
+  update_interval_climate: 5
+  force_miles: true
+"""
+
+import logging
+from datetime import timedelta, datetime
+import time
+import urllib
+import asyncio
+import sys
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect, dispatcher_send)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+
+# Currently waiting on the lib author to review a PR
+# So using my fork for now.
+# https://github.com/jdhorne/pycarwings2/pull/28
+
+REQUIREMENTS = ['https://github.com/BenWoodford/pycarwings2/archive/master.zip'

```

Comment:
We should not merge this until we can point it at the updated pycarwings2 lib on PyPi

Reply:
Noted. I'm still waiting on the author to merge the PR in - something about Python 2 support.